packages/elements/src/number-field/index.ts

```diff
@@ -316,12 +318,24 @@ export class NumberField extends FormFieldElement {
     }
 
     const target = event.target;
+    let direction = Direction.Down;
+    let eventName = '';
 
     if (target === this.spinnerDownEl) {
-      this.onApplyStep(Direction.Down);
+      eventName = 'down-click';
     } else if (target === this.spinnerUpEl) {
-      this.onApplyStep(Direction.Up);
+      eventName = 'up-click';
+      direction = Direction.Up;
```

Comment:
I think we should revise this method to reading easier.

Reply:
How about this one? I'm not sure is it more complex or not.

```typescript
private dispatchSpinnerClickEvent(eventName: string): boolean {
  return this.dispatchEvent(
    new CustomEvent(eventName, {
      cancelable: true
    })
  );
}

protected onSpinnerTap(event: TapEvent): void {
  if (this.disabled || this.readonly || event.defaultPrevented) {
    return;
  }

  const target = event.target;
  if (target === this.spinnerDownEl) {
    this.dispatchSpinnerClickEvent('down-click') && this.onApplyStep(Direction.Down);
  } else if (target === this.spinnerUpEl) {
    this.dispatchSpinnerClickEvent('up-click') && this.onApplyStep(Direction.Up);
  }
}
```

Reply:
That great and easier to read. Updated by your suggestion.